### PR TITLE
Dev bran controller

### PR DESCRIPTION
**Pull request by B08waffles**
Added two features and a test for each. 
CORS has been added server wide on app.ts, going with default settings for now with discussions on whether we go
server wide or route specific for certain differences in settings? 
Rate limits in the form of "express-rate-limit" has been added to app.ts as well, here the rate is set to allow up to 2 request per second, a number that I am happy to change if need be. 

I also added the --verbose flag to the test script for slightly more detail when debugging failed tests.

### CODE REVIEW

File: __test__/integrationTests.test.ts [hunk 1]
```diff
@@ -12,28 +12,46 @@ let expressInstance: Express;
...
-		dbRecordIdIsSessionId: true,
-	});
-	expressInstance = app(store);
-	request = supertest(expressInstance);
+    // The code below initialises the test application in a way which allows the
+    // database and session store to be shut down manually. This is necessary
+    // to avoid the test application from leaking memory and allows Jest to exit
+    // cleanly.
+    store = new PrismaSessionStore(prisma, {
+        checkPeriod: 10 * 60 * 1000, // 1 hour in milliseconds
```
Comment (2016lisali) [inline __test__/integrationTests.test.ts]: should it be 60 *60 *1000 = 1 hour?

File: index.ts [hunk 2]
```diff
@@ -9,11 +9,11 @@ const PORT: string | undefined = process.env.PORT || "3000";
 
 console.log(typeof app);
 const store = new PrismaSessionStore(prisma, {
-	checkPeriod: 10 * 60 * 1000, // 1 hour in milliseconds
-	dbRecordIdIsSessionId: true,
+    checkPeriod: 10 * 60 * 1000, // 1 hour in milliseconds
```
Comment (2016lisali) [inline index.ts]: Should it be 60 * 60 * 1000?

File: app.ts [hunk 3]
```diff
@@ -2,36 +2,52 @@ import express, { Express } from "express";
...
+    // Enable express-rate-limit, for more information on how to use this package
+    // please refer to https://www.npmjs.com/package/express-rate-limit
+    const limiter = rateLimit({
+        windowMs: 1000, // 1000 milliseconds / 1 second
+        max: 2, // Limit each IP to 2 requests per `window` (here, per 1 second)
+        standardHeaders: true, // Return rate limit info in the `RateLimit-*` headers
+        legacyHeaders: false, // Disable the `X-RateLimit-*` headers
+    });
+    server.use(limiter);
+
```
Comment (Jet004) [inline app.ts]: The one thing I would suggest is to add the CORS and rate limit middleware before the session middleware. This way if a request is blocked by either CORS or rate limiting, the session middleware won't run --> this reduces (very slightly) server computations and DB queries. More efficient and reduces server load if there are a lot of requests.

Comment (B08waffles) [reply app.ts]: Good idea! Now that you point it out it makes a lot more sense then leaving it like this, i'll readjust this in the next commit. Thanks for your attention to detail.

File: __test__/integrationTests.test.ts (hunk 1 above)
Comment (Jet004) [reply __test__/integrationTests.test.ts]: You are correct. This is code I wrote in a previous pull request. Are you happy for me to change it in my next update and push it with the logout code? Good job noticing this!

File: index.ts (hunk 2 above)
Comment (Jet004) [reply index.ts]: Same as above comment

File: __test__/integrationTests.test.ts (hunk 1 above)
Comment (2016lisali) [reply __test__/integrationTests.test.ts]: Yep :)

**Review by 2016lisali — APPROVED** [review]
Looks Good :)

File: __test__/integrationTests.test.ts (hunk 1 above)
Comment (Jet004) [reply __test__/integrationTests.test.ts]: This update has already been merged to the development branch 😄

**Review by QiujuYang — APPROVED** [review]

**Review by Jet004 — APPROVED** [review]

**Pull request merged**